### PR TITLE
Adds PE routers as topology nodes

### PR DESCRIPTION
**Pull request by trozet**
Now nodes will be registered as topology nodes. When creating L3VPN the
sites will no longer contain the node information, and instead
reference a node-id which will reference a node in topology.

Signed-off-by: Tim Rozet <tdrozet@gmail.com>

### CODE REVIEW

File: northbound/api/pom.xml [hunk 1]
```diff
@@ -36,6 +36,11 @@ and is available at http://www.eclipse.org/legal/epl-v10.html
             <groupId>org.opendaylight.mdsal.model</groupId>
             <artifactId>ietf-topology</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
```
Comment (shague) [inline northbound/api/pom.xml:40]: any idea what required this?

File: model/ietf-l3vpn-svc/src/main/yang/l3vpn-svc-aug@2017-05-02.yang [hunk 2]
```diff
@@ -111,6 +111,11 @@ module l3vpn-svc-aug {
       description "Provider Edge node-id from network-topology:network-topology/topology/pe/node/<node-id>.";
     }
 
+    leaf ce-node-id {
```
Comment (shague) [inline model/ietf-l3vpn-svc/src/main/yang/l3vpn-svc-aug@2017-05-02.yang:114]: guess we should look at bringing in the newer models at some point as they bring in some of these changes.

File: northbound/api/pom.xml [hunk 3]
```diff
@@ -48,7 +53,8 @@ and is available at http://www.eclipse.org/legal/epl-v10.html
                     <instructions>
                         <Export-Package>
                             org.opendaylight.ansible.northbound.api,
-                            org.opendaylight.yang.gen.v1.urn.opendaylight.ansible.*
+                            org.opendaylight.yang.gen.v1.urn.opendaylight.ansible.*,
+                            org.opendaylight.yang.gen.v1.urn.tbd.params.xml.ns.yang.network.topology.aug.*
```
Comment (shague) [inline northbound/api/pom.xml:57]: should make sure this is only exporting augmentations we defined. check the manifest file in the jar to see what it says is exported.

File: northbound/api/src/main/java/org/opendaylight/ansible/northbound/api/AnsibleTopology.java [hunk 4]
```diff
@@ -28,5 +50,108 @@ private AnsibleTopology() {}
             .child(Topology.class, new TopologyKey(ANSIBLE_TOPOLOGY_ID))
             .child(Node.class);
 
+    public static final InstanceIdentifier<Link> ANSIBLE_LINK_PATH =
+            InstanceIdentifier.create(NetworkTopology.class)
+                    .child(Topology.class, new TopologyKey(ANSIBLE_TOPOLOGY_ID))
+                    .child(Link.class);
 
+    public static Node getNode(NodeId nodeId, DataBroker dataBroker) {
```
Comment (shague) [inline northbound/api/src/main/java/org/opendaylight/ansible/northbound/api/AnsibleTopology.java:58]: reason to have all this static? Would be better to add a topology class and then inject it to this class for use.

File: northbound/api/pom.xml (hunk 1 above)
Comment (trozet) [reply northbound/api/pom.xml:40]: Look in AnsibleTopology, it gives the Pair object:
https://github.com/trozet/opendaylight-ansible/blob/add_topology/northbound/api/src/main/java/org/opendaylight/ansible/northbound/api/AnsibleTopology.java#L14

File: model/ietf-l3vpn-svc/src/main/yang/l3vpn-svc-aug@2017-05-02.yang (hunk 2 above)
Comment (trozet) [reply model/ietf-l3vpn-svc/src/main/yang/l3vpn-svc-aug@2017-05-02.yang:114]: yeah, this wont be committed upstream so it was really just for the demo

File: northbound/api/pom.xml (hunk 3 above)
Comment (trozet) [reply northbound/api/pom.xml:57]: this is just for the demo, I don't think it matters as it wont be committed upstream and clash with any other exports

File: northbound/api/src/main/java/org/opendaylight/ansible/northbound/api/AnsibleTopology.java (hunk 4 above)
Comment (trozet) [reply northbound/api/src/main/java/org/opendaylight/ansible/northbound/api/AnsibleTopology.java:58]: No reason to have it static. You are saying we should initialize the flow:1 topology and create an instance of this on boot? Then just reference that object and remove the statics?

**Pull request merged**